dataclients/kubernetes/main_test.go

```diff
@@ -0,0 +1,11 @@
+package kubernetes_test
+
+import (
+	"testing"
+
+	"go.uber.org/goleak"
+)
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
+}
```

Comment:
I am not completely sure what this does but https://github.com/uber-go/goleak#quick-start seems to suggest it will put it into all tests. I validated with a known broken test (proxy package that has test with go-redis that starts heartbeat which is not closed) and it does not seem to matter if we are in the same package `kubernetes` here or another package like `kubernetes_test`.

Reply:
Then let's remove it if it doesn't matter?

Reply:
I think this test injects the test for every test in the package